add tests that schemas are valid

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30356 [JIT] polish up overloads on free functions
* **#30749 add tests that schemas are valid**

Add check to schemas that the schema is sane.

I removed the defaults from symbolic_script because they were in some cases wrong and don't actually do anything. At the point they're invoked the forward should already have matched all arguments.

Differential Revision: [D18864775](https://our.internmc.facebook.com/intern/diff/D18864775)

Commit:
add tests that schemas are valid

[ghstack-poisoned]

ailzhang reviewed (approved):
Stamping the symbolic_script changes. Will leave the review for the remaining changes to @suo ;).

suo commented:
Yeah seems good. I don't see a CMakelists/build_variables.py change though, you'll probably need that

eellison commented:
> Yeah seems good. I don't see a CMakelists/build_variables.py change though, you'll probably need that

I ctrl-F'd other cpp files in that directory and didn't see anything

Commit:
Update on "add tests that schemas are valid"


Add check to schemas that the schema is sane.

I removed the defaults from symbolic_script because they were in some cases wrong and don't actually do anything. At the point they're invoked the forward should already have matched all arguments.


[ghstack-poisoned]

Commit:
Update on "add tests that schemas are valid"


Add check to schemas that the schema is sane.

I removed the defaults from symbolic_script because they were in some cases wrong and don't actually do anything. At the point they're invoked the forward should already have matched all arguments.


[ghstack-poisoned]
